packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -367,6 +345,33 @@ class OverlayState extends State<Overlay> with TickerProviderStateMixin {
     });
   }
 
+  void _assertEntriesInsertedPosition(OverlayEntry above, OverlayEntry below,
+      {Iterable<OverlayEntry> newEntries}) {
+    assert(
+      above == null || below == null,
+      'Only one of `above` and `below` may be specified.',
+    );
+    if (newEntries == null) {
+      assert(
+        above == null || (above._overlay == this && _entries.contains(above)),
+        'The provided entry for `above` is not present in the Overlay.',
+      );
+      assert(
+        below == null || (below._overlay == this && _entries.contains(below)),
+        'The provided entry for `below` is not present in the Overlay.',
+      );
+    } else {
+      assert(
+        above == null || (above._overlay == this && _entries.contains(above) && newEntries.contains(above)),
+        'The entry used for `above` must be in the Overlay and in the `newEntriesList`.'
+      );
+      assert(
+        below == null || (below._overlay == this && _entries.contains(below) && newEntries.contains(below)),
+        'The entry used for `below` must be in the Overlay and in the `newEntriesList`.'
+      );
+    }
```

Comment:
While there's less duplication, the code is actually a lot harder to read now than it was before with the if/else branches. I think I'd prefer the old style.